src/eeprom.c

```diff
@@ -0,0 +1,102 @@
+#include "eeprom.h"
+#include "formatting.h"
+
+#define BYTES_PER_WORD 4
+
+/**
+ * @brief Function to get the words of a number of bytes.
+ *
+ * It includes garbage bytes if needed.
+ *
+ * @param[in] num_bytes Number of bytes.
+ * @return Number of words.
+ */
+static uint16_t bytes_to_words(uint16_t num_bytes)
+{
+	uint16_t num_words = num_bytes / BYTES_PER_WORD;
+
+	if ((num_bytes % BYTES_PER_WORD) > 0)
+		++num_words;
+
+	return num_words;
+}
+/**
+ * @brief Function to read EEPROM data from a specific address.
+ *
+ * @param[in] start_address Address to read from.
+ * @param[in] num_elements Number of bytes to be read.
+ * @param[out] output_data Pointer to a buffer to save the read data.
+ */
+void eeprom_read_data(uint32_t start_address, uint16_t num_bytes,
+		      uint8_t *output_data)
+{
+	uint16_t iter;
+	uint32_t *memory_ptr = (uint32_t *)start_address;
+
+	for (iter = 0; iter < bytes_to_words(num_bytes); iter++) {
+		*(uint32_t *)output_data = *(memory_ptr + iter);
+		output_data += BYTES_PER_WORD;
+	}
+}
```

Comment:
I think `output_data` should be `uint32_t`:

- First, it is a bit more consistent within the loop, as you would not need `BYTES_PER_WORD` and could increment `output_data++` or use `(output_data + iter)` (just like you do with `memory_ptr`.
- Second, probably more important, it is prone to out-of-bounds bugs: you may pass an `uint8_t *output_data` of size 7 because you want to read 7 bytes. But then the function will write 8 bytes instead, writing out of bounds and potentially breaking the system. If you use `uint32_t` I think it is less likely to hit that issue?

Reply:
80aff94

Reply:
* Issue #328 to change output_data 
* I modified the implementation of the maze_is_saved function to avoid an out of boundary bug, so the verification is done with 4 bytes. As maze structure size is 256, there are no bugs at the moment.